dcicutils/misc_utils.py

```diff
@@ -11,6 +13,54 @@
 PRINT = print
 
 
+class VirtualApp(TestApp):
+    """ Wrapper class for TestApp, which we use a handler to the Encoded Application where we can submit
+        requests to it simulating a number of conditions, including permissions.
+    """
+    logging.basicConfig()
+
+
+    def __init__(self, app, environ):
+        """ Builds an encoded application, allowing you to submit requests to an encoded application
+
+        :param app: return value of get_app(config_uri, app_name)
+        :param environ: options to pass to the application. Usually permissions.
+        """
+        self.virtual_app = TestApp(app, environ)
```

Comment:
If you're going to make it have an attribute that holds a TestApp (and that's not a terrible plan, just a design choice), then you don't want to also inherit from it. In that case, you'd be the instance itself, inheriting its `__init__` method from `TestApp` so it doesn't need a new one unless it's going to rewrite the instantiation protocol.